public/app/plugins/datasource/prometheus/querybuilder/components/promQail/state/helpers.ts

```diff
@@ -40,15 +39,51 @@ interface TemplateSearchResult {
   promql: string | null;
 }
 
-function getExplainMessage({
-  documentation,
-  metricType,
-  description,
-  query,
-}: ExplainUserPromptParams): llms.openai.Message[] {
+// export function getExplainMessage({
+//   documentation,
+//   metricName,
+//   metricType,
+//   metricMetadata,
+//   query,
+// }: ExplainUserPromptParams): llms.openai.Message[] {
+//   return [
+//     { role: 'system', content: ExplainSystemPrompt },
+//     { role: 'user', content: GetExplainUserPrompt({ documentation, metricName, metricType, metricMetadata, query }) },
+//   ];
+// }
+
+export function getExplainMessage(
+  query: string,
+  metric: string,
+  datasource: PrometheusDatasource,
+): llms.openai.Message[] {
+
+  let metricMetadata = '';
+  let metricType = '';
+
+  const pvq = buildVisualQueryFromString(query);
+
+  if (datasource.languageProvider.metricsMetadata) {
+    metricType = getMetadataType(metric, datasource.languageProvider.metricsMetadata!) ?? '';
```

Comment:
`datasource.languageProvider.metricsMetadata` shouldn't need the `!` in this branch.